src/libertem/io/dataset/raw.py

```diff
@@ -116,19 +122,39 @@ def __init__(self, path, scan_size, dtype, detector_size=None, enable_direct=Fal
                 raise ValueError("RawFileDataSet can't crop detector anymore, "
                                  "please use EMPAD DataSet")
             detector_size = crop_detector_to
-
-        if detector_size is None:
-            raise TypeError("missing 1 required argument: 'detector_size'")
-
+        if sig_shape is None and detector_size is None:
+            raise TypeError("missing 1 required argument: 'sig_shape'")
+        self._nav_shape = tuple(nav_shape) if nav_shape else nav_shape
+        self._sig_shape = tuple(sig_shape) if sig_shape else sig_shape
+        self._sync_offset = sync_offset
+        # handle backwards-compatability:
+        if scan_size is not None:
+            warnings.warn(
+                "scan_size argument is ignored and will be removed after 0.6.0",
```

Comment:
Again should point to its replacement and clarify that it is not ignored but used for `nav_shape` right now.